tracer/src/Datadog.Trace/Debugger/Caching/DefaultEnvironmentChecker.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="DefaultEnvironmentChecker.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Threading;
+
+namespace Datadog.Trace.Debugger.Caching;
+
+internal class DefaultEnvironmentChecker : IEnvironmentChecker
+{
+    private static readonly Lazy<DefaultEnvironmentChecker> _instance = new Lazy<DefaultEnvironmentChecker>(() => new DefaultEnvironmentChecker(), LazyThreadSafetyMode.ExecutionAndPublication);
+    private readonly bool _isServerlessEnvironment;
+
+    private DefaultEnvironmentChecker()
+    {
+        _isServerlessEnvironment = CheckServerlessEnvironment();
+    }
+
+    public static DefaultEnvironmentChecker Instance => _instance.Value;
+
+    public bool IsServerlessEnvironment()
+    {
+        return _isServerlessEnvironment;
+    }
```

Comment:
Should this be a property instead of a method?

Reply:
Thanks! Fixed in 93ce326e53379d4935cf896c25e5600ea89b4164